.github/workflows/test.yml

```diff
@@ -39,3 +39,21 @@ jobs:
       uses: golangci/golangci-lint-action@v3
       with:
         version: v1.54.1
+
+  goreleaser:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v4
+      with:
+        fetch-depth: 0
+
+    - name: Set up Go
+      uses: actions/setup-go@v4
+      with:
+        go-version: 1.21.1
+
+    - name: Run GoReleaser
+      uses: goreleaser/goreleaser-action@v5
+      with:
+        version: latest
+        args: release --clean --snapshot
```

Comment:
I have a feeling this is going to want to have the `--skip-publish` flag too? https://goreleaser.com/quick-start/#release-flags

Reply:
Those docs are actually out of date:

```
  • DEPRECATED: --skip-publish was deprecated in favor of --skip=publish, check https://goreleaser.com/deprecations#-skip for more details
  • skipping announce and publish...
```

Check out the `--snapshot` flag:

```
Releases the current project

Usage:
  goreleaser release [flags]

Aliases:
  release, r

Flags:
      --auto-snapshot                Automatically sets --snapshot if the repository is dirty
      --clean                        Removes the dist folder
  -f, --config string                Load configuration from file
      --fail-fast                    Whether to abort the release publishing on the first error
  -h, --help                         help for release
  -p, --parallelism int              Amount tasks to run concurrently (default: number of CPUs)
      --release-footer string        Load custom release notes footer from a markdown file
      --release-footer-tmpl string   Load custom release notes footer from a templated markdown file (overrides --release-footer)
      --release-header string        Load custom release notes header from a markdown file
      --release-header-tmpl string   Load custom release notes header from a templated markdown file (overrides --release-header)
      --release-notes string         Load custom release notes from a markdown file (will skip GoReleaser changelog generation)
      --release-notes-tmpl string    Load custom release notes from a templated markdown file (overrides --release-notes)
      --skip strings                 Skip the given options (valid options are announce, before, docker, ko, publish, sbom, sign, validate)
      --snapshot                     Generate an unversioned snapshot release, skipping all validations and without publishing any artifacts (implies --skip=announce,publish,validate)
      --timeout duration             Timeout to the entire release process (default 30m0s)

Global Flags:
      --debug     Enable verbose mode
      --verbose   Enable verbose mode
```

And docs: https://goreleaser.com/customization/snapshots/

> Note that the idea behind GoReleaser's snapshots is for local builds or to validate your build on the CI pipeline. Artifacts won't be uploaded and will only be generated into the dist folder.

Reply:
Also quick pros and cons list for the dry run:

Pros:
- It caught an actual issue with our goreleaser I would not have otherwise found

Cons:
- It took 26 minutes to run and we have a finite number of CI minutes to spend

I'd probably say it's too expensive to run on every code change. Maybe leave it in, make it optional for merge, and trigger it manually if we believe that a PR might break it? I'll wait for your opinion before I trigger another 26 minute CI pipeline.

Reply:
Wow that is a lot of time. How many free minutes do we have? I would say we can make it optional, not like it's gonna be broken like this that often

Reply:
2,000 minutes/month per organization (not per repo). 

If we take the last PR as an example, this would have run 7 times from initial push to merge, taking up a total of ~182 minutes, or just under 10% of our organization's monthly allowed Actions minutes for one change. So uh yeah definitely not going to want to run this on every PR. 

Unfortunately it doesn't look like GitHub Actions supports a straightforward workflow for "register job on event, click to trigger". We could use [environments](https://docs.github.com/en/actions/deployment/targeting-different-environments/using-environments-for-deployment), which I think will proxy the behavior we're looking for, but it feels like fitting a square peg into a round hole for me. 

So I dropped the check, and we can revisit if it becomes a problem over time.

Reply:
I am cool with a longer pipeline. lets keep it in.